packages/@uppy/provider-views/src/utils/PartialTreeUtils/afterOpenFolder.ts

```diff
@@ -13,8 +13,16 @@ const afterOpenFolder = (
   currentPagePath: string | null,
   validateSingleFile: (file: CompanionFile) => string | null,
 ): PartialTree => {
-  const discoveredFolders = discoveredItems.filter((i) => i.isFolder === true)
-  const discoveredFiles = discoveredItems.filter((i) => i.isFolder === false)
+  const discoveredFolders = discoveredItems.filter(
```

Comment:
This is an additional check to fix an edge case where a folder or file added to the partialTree through search mode (by navigating into it or selecting it) would get duplicated when manually navigating to its parent via the path or breadcrumbs, causing another folder with the same ID to be added through the list API.